Kernel/Arch/x86/common/ProcessorInfo.cpp

```diff
@@ -56,14 +56,22 @@ ProcessorInfo::ProcessorInfo(Processor& processor)
     u32 max_extended_leaf = CPUID(0x80000000).eax();
 
     if (max_extended_leaf >= 0x80000004) {
-        alignas(u32) char buffer[48];
-        u32* bufptr = reinterpret_cast<u32*>(buffer);
+        char buffer[49];
+        buffer[48] = '\0';
+        char* bufptr = buffer;
         auto copy_brand_string_part_to_buffer = [&](u32 i) {
+            auto store_u32 = [](char* target, u32 value) {
+                target[0] = value & 0xFF;
+                target[1] = (value >> 8) & 0xFF;
+                target[2] = (value >> 16) & 0xFF;
+                target[3] = (value >> 24) & 0xFF;
+            };
```

Comment:
this is reimplementing ByteBuffer::store<u32>

standrd endiannes on x86 is little endian, so memcpy is eqivalent and the compiler will generate approximately the same code

Reply:
`s/ByteBuffer/ByteReader/` :P